src/main/java/ch/njol/skript/effects/EffOpenInventory.java

```diff
@@ -34,125 +36,220 @@
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Effect;
 import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.Literal;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.skript.util.Version;
 import ch.njol.util.Kleenean;
 
 @Name("Open/Close Inventory")
-@Description({"Opens an inventory to a player. The player can then access and modify the inventory as if it was a chest that he just opened.",
-		"Please note that currently 'show' and 'open' have the same effect, but 'show' will eventually show an unmodifiable view of the inventory in the future."})
-@Examples({"show the victim's inventory to the player",
-		"open the player's inventory for the player"})
-@Since("2.0, 2.1.1 (closing), 2.2-Fixes-V10 (anvil), 2.4 (hopper, dropper, dispenser")
+@Description({
+	"Opens an inventory to a player. The player can then access and modify the inventory as if it was a chest that they just opened.",
+	"Note that 'show' and 'open' have different effects, 'show' will show just a view of the inventory.",
+	"Whereas 'open' will attempt to make an inventory real and usable. Like a workbench allowing recipes to work."
+})
+@Examples({
+	"show crafting table to player #unmodifiable, use open instead to allow for recipes to work",
+	"open a crafting table to the player",
+	"open a loom to the player",
+	"open the player's inventory for the player"
+})
+@Since("2.0, 2.1.1 (closing), 2.2-Fixes-V10 (anvil), INSERT VERSION (enchanting, cartography, grindstone, loom)")
 public class EffOpenInventory extends Effect {
-	
-	private final static int WORKBENCH = 0, CHEST = 1, ANVIL = 2, HOPPER = 3, DROPPER = 4, DISPENSER = 5;
-	
+
+	private static enum OpenableInventorySyntax {
+
+		ANVIL("anvil"),
+		CARTOGRAPHY("cartography [table]", Skript.methodExists(HumanEntity.class, "openCartographyTable", Location.class, boolean.class),
+				"Opening a cartography table inventory requires PaperSpigot."),
+		ENCHANTING("enchant(ment|ing) [table]", new Version(1, 14)),
+		GRINDSTONE("grindstone", Skript.methodExists(HumanEntity.class, "openGrindstone", Location.class, boolean.class),
+				"Opening a grindstone inventory requires PaperSpigot."),
+		LOOM("loom", Skript.methodExists(HumanEntity.class, "openLoom", Location.class, boolean.class),
+				"Opening a loom inventory requires PaperSpigot."),
+		SMITHING("smithing [table]", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
+				"Opening a smithing table inventory requires PaperSpigot."),
+		STONECUTTER("stone[ ]cutter", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
+				"Opening a stone cutter inventory requires PaperSpigot."),
+		WORKBENCH("(crafting [table]|workbench)");
+
+		@Nullable
+		private String methodError;
```

Comment:
```suggestion
		private @Nullable String methodError;
```
all the other annotations like this in this class should be inlined too